pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -63,22 +63,21 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
     {"name": "TestLabelPreference",      "weight": 4, "argument": {"labelPreference": {"label": "bar", "presence":true}}}
   ]
 }`,
-			ExpectedPolicy: schedulerapi.Policy{
-				Predicates: []schedulerapi.PredicatePolicy{
-					{Name: "MatchNodeSelector"},
-					{Name: "PodFitsResources"},
-					{Name: "PodFitsPorts"},
-					{Name: "NoDiskConflict"},
-					{Name: "TestServiceAffinity", Argument: &schedulerapi.PredicateArgument{ServiceAffinity: &schedulerapi.ServiceAffinity{Labels: []string{"region"}}}},
-					{Name: "TestLabelsPresence", Argument: &schedulerapi.PredicateArgument{LabelsPresence: &schedulerapi.LabelsPresence{Labels: []string{"foo"}, Presence: true}}},
-				},
-				Priorities: []schedulerapi.PriorityPolicy{
-					{Name: "LeastRequestedPriority", Weight: 1},
-					{Name: "ServiceSpreadingPriority", Weight: 2},
-					{Name: "TestServiceAntiAffinity", Weight: 3, Argument: &schedulerapi.PriorityArgument{ServiceAntiAffinity: &schedulerapi.ServiceAntiAffinity{Label: "zone"}}},
-					{Name: "TestLabelPreference", Weight: 4, Argument: &schedulerapi.PriorityArgument{LabelPreference: &schedulerapi.LabelPreference{Label: "bar", Presence: true}}},
-				},
-			},
+			wantPredicates: sets.NewString(
+				"CheckNodeCondition", //mandatory predicate
```

Comment:
why is this added to the compatibility test? it was added only after 1.9: https://github.com/kubernetes/kubernetes/blob/8993fbc543c18e73668793b5d5e234c0a136735c/CHANGELOG-1.9.md

Reply:
that's because the scheduler will add it to the predicates, so if we don't add it there schedPredicates and wantPredicates won't match.